lib/test/Test/Pos/Txp/Toil/UtxoSpec.hs

```diff
@@ -435,24 +446,39 @@ scriptTxSpec = describe "script transactions" $ do
 -- | Script transaction should pass the check and return 'Right'.
 txShouldSucceed :: Either ToilVerFailure () -> Expectation
 txShouldSucceed test = whenLeft test $ \x ->
-    expectationFailure $ "unexpected failure: " <> show x
+    expectationFailure $ toString $ "unexpected failure: " <> pretty x
 
 -- | Transaction should fail with a 'ToilWitnessDoesntMatch' error.
 txShouldFailWithWitnessMismatch :: Either ToilVerFailure () -> Expectation
 txShouldFailWithWitnessMismatch = \case
     Left ToilWitnessDoesntMatch{} -> pass
-    other -> expectationFailure $
+    other -> expectationFailure $ toString $
         "expected: Left ToilWitnessDoesntMatch{..}\n" <>
-        " but got: " <> show other
+        " but got: " <> either pretty show other
 
 -- | Transaction should fail with a Plutus error.
 txShouldFailWithPlutus :: Either ToilVerFailure () -> PlutusError -> Expectation
 txShouldFailWithPlutus res err = case res of
     Left ToilInvalidWitness{..}
         | tiwReason == WitnessScriptError err -> pass
-        | otherwise -> expectationFailure $
-              "expected: " <> show (WitnessScriptError err) <> "\n" <>
-              " but got: " <> show tiwReason
-    other -> expectationFailure $
-        "expected: Left ...: " <> show (WitnessScriptError err) <> "\n" <>
-        " but got: " <> show other
+        | otherwise -> expectationFailure $ toString $
+              "expected: " <> pretty (WitnessScriptError err) <> "\n" <>
+              " but got: " <> pretty tiwReason
+    other -> expectationFailure $ toString $
+        "expected: Left ...: " <> pretty (WitnessScriptError err) <> "\n" <>
+        " but got: " <> either pretty show other
+
+-- | Transaction should fail with a 'ToilInvalidWitness' specifically 'WitnessAdoptedScriptVerMismatch' error.
+txShouldFailWithInvalidAdoptedScriptVer :: Either ToilVerFailure () -> Expectation
+txShouldFailWithInvalidAdoptedScriptVer = \case
+    Left (ToilInvalidWitness _ _ c) -> case c of
+        WitnessAdoptedScriptVerMismatch _ _ -> pass
+        WitnessScriptVerMismatch _ _ -> expectationFailure $
+            "expected: WitnessAdoptedScriptVerMismatch\n" <>
+            " but got: WitnessScriptVerMismatch"
```

Comment:
Why are you special-casing this branch? I don't like that the two fields of `WitnessScriptVerMismatch` won't be shown when tests fail.